src/otx/v2/adapters/torch/mmengine/mmaction/dataset.py

```diff
@@ -0,0 +1,255 @@
+                Defaults to None.
+            train_ann_files (str | None, optional): Location of the annotation file for the dataset
+                to be used for training. Defaults to None.
+            val_data_roots (str | None, optional): The root address of the dataset
+                to be used for validation. Defaults to None.
+            val_ann_files (str | None, optional): Location of the annotation file for the dataset
+                to be used for validation. Defaults to None.
+            test_data_roots (str | None, optional): The root address of the dataset
+                to be used for testing. Defaults to None.
+            test_ann_files (str | None, optional): Location of the annotation file for the dataset
+                to be used for testing. Defaults to None.
+            unlabeled_data_roots (str | None, optional): The root address of the unlabeled dataset
+                to be used for training. Defaults to None.
+            unlabeled_file_list (str | None, optional): The file where the list of unlabeled images is declared.
+                Defaults to None.
+            data_format (str | None, optional): The format of the dataset. Defaults to None.
+        """
+        super().__init__(
+            task,
+            train_type,
+            train_data_roots,
+            train_ann_files,
+            val_data_roots,
+            val_ann_files,
+            test_data_roots,
+            test_ann_files,
+            unlabeled_data_roots,
+            unlabeled_file_list,
+            data_format,
+        )
+        self.scope = "mmaction"
+        self.dataset_registry = MMActionRegistry().get("dataset")
+
+    def _initialize(self) -> None:
+        self.set_datumaro_adapters()  # Set self.dataset_entity & self.label_schema
+        self.base_dataset = self._get_sub_task_dataset()
+        self.initialize = True
+
+    def _get_sub_task_dataset(self) -> TorchDataset:
+        # ruff: noqa: TD002, TD003, FIX002
+        #TODO: will be added the OTXActionDetDataset
+        return OTXActionClsDataset
+
+    def _build_dataset(
+        self,
+        subset: str,
+        pipeline: list | None = None,
+        config: dict | None = None,
+    ) -> TorchDataset | None:
+        """Builds a TorchDataset object for the given subset using the specified pipeline and configuration.
+
+        Args:
+            subset (str): The subset to build the dataset for.
+            pipeline (list | None, optional): The pipeline to use for the dataset.
+                Defaults to None.
+            config (dict | None, optional): The configuration to use for the dataset.
+                Defaults to None.
+
+        Returns:
+            TorchDataset | None: The built TorchDataset object, or None if the dataset is empty.
+        """
+        if pipeline is None:
+            pipeline = get_default_pipeline(subset)
+        return super()._build_dataset(subset, pipeline, config)
+
+    def _build_dataloader(
```

Comment:
can't it be removed?

Reply:
I think this is for using specific collate_fn and I'll handle this part in phase 2(unifying engine.py)